ibazel/main.go

```diff
@@ -53,30 +66,16 @@ func main() {
 		return
 	}
 
-	b = bazel.New()
+	b := bazel.New()
 
 	command := os.Args[1]
 	target := os.Args[2]
 
-	query := fmt.Sprintf("kind('source file', deps('%s'))", target)
+	var buildFileWatcher *fsnotify.Watcher
```

Comment:
One problem in my experience is that `*fsnotify.Watcher.Events` is not buffered. That means it'll block once some event happens, and the interleaving of you reading from the channel and events arriving is complex and can lead to bouncy behaviour (if not races).

What you can is create your own channel with appropriate buffering (10 items or so) and a goroutine that continuously reads fsnotify's channel and writes into the buffer.

Then to debounce, the simplest approach I can think of is to just `time.Sleep(100 * time.Millisecond)` (or whatever number, 20 ms might be enough, too), then synchronously read all events of the buffered channel, i.e. deplete it. That gives a relatively straight forward algorithm that should cover most eventualities (unless you e.g. have a ton of events arriving).